web/src/app/modules/shared/components/presentation/datagrid/datagrid.component.ts

```diff
@@ -111,17 +114,27 @@ export class DatagridComponent
 
     return rows.map(row => {
       let actions: GridAction[] = [];
+      let detailView: View;
+      let replace: boolean;
 
       if (row.hasOwnProperty('_action')) {
         actions = (row._action as GridActionsView).config.actions;
       }
 
+      if (row.hasOwnProperty('_expand')) {
+        const erv = (row._expand as ExpandableRowDetailView).config;
+        detailView = erv.body;
+        replace = erv.replace;
+      }
+
       const isDeleted = !!row._isDeleted;
 
       return {
         data: row,
         actions,
         isDeleted,
+        detailView,
```

Comment:
`TableRowWithMetadata` needs to be updated with new fields

Reply:
What new fields? An expandable row exists as a `"_expand":View` pair on TableRow.

Reply:
Sorry, that was not clear... if `getRowsWithMetadata()` and `rowsWithMetadata` are declared as `TableRowWithMetadata[]`, then `TableRowWithMetadata` now needs `detailView` and `replace` fields added.